modules/core/src/main/java/org/apache/ignite/internal/maintenance/MaintenanceProcessor.java

```diff
@@ -376,6 +381,10 @@ private void startShutdownThread() {
                 while (true) {
                     if (activeTasks.isEmpty() && Ignition.state(igniteInstanceName) == IgniteState.STARTED) {
                         G.stop(igniteInstanceName, false);
+
+                        if (mntcModeExitCode != 0)
+                            System.exit(mntcModeExitCode);
```

Comment:
I am not sure, that it works fine. I would try to create an override for G#stop to have optional exit code. But I am ok to have as it is for now as experimental option.